src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java

```diff
@@ -91,6 +127,170 @@ private EventProcessorIdentifier eventProcessorId(String eventProcessorName, Str
                                        .build();
     }
 
+    @Override
+    public CompletableFuture<Void> createOrUpdateUser(CreateOrUpdateUserRequest request)  {
+        FutureStreamObserver<Empty> responseObserver = new FutureStreamObserver<>(null);
+        userServiceStub.createOrUpdateUser(request, responseObserver);
+        return responseObserver.thenAccept(empty -> {
+        });
+    }
+
+    @Override
+    public CompletableFuture<List<UserOverview>> getAllUsers() {
+        FutureListStreamObserver<UserOverview> responseObserver = new FutureListStreamObserver<>();
+        userServiceStub.getUsers(Empty.newBuilder().build(),responseObserver);
```

Comment:
Curious: why do we need to send an empty object to get all the users? How does the receiving end know that "empty = get all users"?

Reply:
I didnt find any other way to return stream of something, when there is no specific request to send. There is only one similar example in Axon Server internal API, it also uses EmptyRequest